Sources/SemanticAnalyzer/TypeChecker.swift

```diff
@@ -177,7 +177,28 @@ public struct TypeChecker: ASTPass {
   }
 
   public func process(subscriptExpression: SubscriptExpression, passContext: ASTPassContext) -> ASTPassResult<SubscriptExpression> {
-    return ASTPassResult(element: subscriptExpression, diagnostics: [], passContext: passContext)
+    var diagnostics = [Diagnostic]()
+    let environment = passContext.environment!
+    let typeIdentifier = passContext.enclosingTypeIdentifier!
+    let scopeContext = passContext.scopeContext!
+    
+    let identifierType = environment.type(of: .identifier(subscriptExpression.baseIdentifier), enclosingType: typeIdentifier.name, scopeContext: scopeContext)
+    
+    var expectedType: Type.RawType
+    
+    switch identifierType {
+    case .arrayType (_): expectedType = .basicType(.int)
+    case .fixedSizeArrayType(_): expectedType = .basicType(.int)
+    case .dictionaryType(let keyType, _): expectedType = keyType
+    default: fatalError("Invalid identifier type \(identifierType) for subscript expression")
```

Comment:
This means the identifier is not subscriptable right? For example, attempting to do `a[2]` when `a` is a `Bool`. I think there should be a Flint error rather than a `fatalError`

Reply:
Agreed